packages/fastify-siwe/src/api.test.ts

```diff
@@ -1,13 +1,28 @@
 import { expect } from 'chai'
+import createFastify, { FastifyInstance } from 'fastify'
 import { MockProvider } from 'ethereum-waffle'
-import 'jsdom-global/register'
 import { SiweMessage } from 'siwe'
-import { mock } from './mockApi'
+import { siwePlugin } from './plugin'
+import { InMemoryStore } from './InMemoryStore'
+import cookie from '@fastify/cookie'
+import { registerSiweRoutes } from './registerSiweRoutes'
+import { Wallet } from 'ethers'
 
-describe('signInWithEthereum', () => {
-  const app = mock()
-  const provider = new MockProvider({ ganacheOptions: { chain: { chainId: 1 } } as any })
-  const signer = provider.getWallets()[0]
+describe('Fastify with SIWE API', () => {
```

Comment:
Add a test where you re-use the same nonce to login twice with a different account.